include/tudat/astro/system_models/selfShadowing.h

```diff
@@ -0,0 +1,491 @@
+/*    Copyright (c) 2010-2019, Delft University of Technology
+*    All rigths reserved
+*
+*    This file is part of the Tudat. Redistribution and use in source and
+*    binary forms, with or without modification, are permitted exclusively
+*    under the terms of the Modified BSD license. You should have received
+*    a copy of the license with this file. If not, please or visit:
+*    http://tudat.tudelft.nl/LICENSE.
+*/
+#ifndef TUDAT_SELFSHADOWING_H

```

Comment:
Could you add references to the relevant equations in literature you implement here (to at least make clear how/which algorithm is implemented)?